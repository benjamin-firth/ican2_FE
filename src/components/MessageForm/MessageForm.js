import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchData } from '../../utils/apiCalls';
import { loadMessages, addMessage } from '../../actions';
import './MessageForm.scss';
import PropTypes from 'prop-types';

const MessageForm = ({ otherMessenger }) => {
  const currentUser = useSelector(state => state.currentUser);
  const dispatch = useDispatch();
  const [messageToSend, setMessageToSend] = useState('');

  const sendMessage = () => {

    const body = {
      query: "mutation {\n createMessage(input: {\n senderId: \"" + currentUser.id + "\"\n recipientId: \"" + otherMessenger.id + "\"\n body: \"" + messageToSend + "\"\n}) {\n message {\n body \n userId\n conversation {\n recipientId\n }\n }\n }\n }",
      variables: {}
    };

    fetchData(body)
      .then (data => dispatch(addMessage(data.data.createMessage.message)))
      .catch(error => error)
    setMessageToSend('');
  };

  const loadMessages = () => {
    fetchData({"query": "{messages(sender: \""+ currentUser.id + "\", recipient: \""+ otherMessenger.id + "\") {body read userId}}"})
    .then(data => dispatch(loadMessages({
      otherMessenger: {
        id: otherMessenger.id,
        name: otherMessenger.name,
        pic: otherMessenger.pic
      },
      messages: data.data.messages
    })))
    .catch(error => error)
  }

  useEffect(() => loadMessages(), []);

  return (
    <div className='message-form-container'>
      <p className='new-message-tag'>new message:</p>
      <form type="reset" className='message-form'>
        <textarea id='stateMessageToSend' value={messageToSend} onChange={e => setMessageToSend(e.target.value)} />
        <button type='button' id='messageButton' onClick={() => sendMessage()}>send message</button>
      </form>
    </div>
  );
}

MessageForm.propTypes = {
  otherMessenger: PropTypes.object
}

export default MessageForm;
