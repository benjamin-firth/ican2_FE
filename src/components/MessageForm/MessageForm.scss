@import '../../variables.scss';

.message-form {
  display: flex;
  justify-content: center;
  align-items: center;

  textarea {
    height: 8vw;
    width: 50vw;
    border: none;
    font-size: 1.4vw;
    padding: 1vw;
  }

  button {
    border: 5px solid $secondarycolor;
    margin: 0;
    margin-left: 1vw;
  }
}

.new-message-tag {
  font-size: 1.8vw;
  font-family: $secondaryfont;
  color: $secondarycolor;
  margin: 1vw 0 .3vw .2vw;
}

@media only screen and (max-width: 400px) {
  .message-form-container {
    width: 90%;
    margin: auto;
  }

  .message-form {
    flex-direction: column;
    justify-content: center;
    align-items: center;

    textarea {
      height: 25vw;
      width: 100%;
      border: none;
      font-size: 1.4vw;
      padding: 1vw;
      margin: 2vw 0;
    }

    button {
      font-size: 6vw;
      margin-left: 1vw;
      width: auto;
      margin-bottom: 3vw;
    }
  }

  .new-message-tag {
    font-size: 7vw;
  }
}
