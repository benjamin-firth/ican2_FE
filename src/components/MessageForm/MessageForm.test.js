import React from 'react';
import { shallow } from 'enzyme';
import MessageForm from './MessageForm';

const mockCurrentUser = {
  id: 3,
  name: 'pants'
}

const mockOtherMessenger = {
  id: 4
}

jest.mock("react-redux", () => ({
  useSelector: () => mockCurrentUser,
  useDispatch: () => jest.fn()
}));

describe('MessageForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <MessageForm otherMessenger={mockOtherMessenger}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
