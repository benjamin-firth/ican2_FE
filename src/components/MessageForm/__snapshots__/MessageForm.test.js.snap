// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm should match the snapshot 1`] = `
<div>
  <p
    className="new-message-tag"
  >
    new message:
  </p>
  <form
    className="message-form"
    type="reset"
  >
    <textarea
      onChange={[Function]}
      value=""
    />
    <button
      onClick={[Function]}
    >
      send message
    </button>
  </form>
</div>
`;
